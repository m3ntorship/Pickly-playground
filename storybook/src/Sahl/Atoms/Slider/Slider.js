import React from "react";
import PropTypes from "prop-types";

export default function Slider({ progress }) {
  let completeClass = progress < 100 ? "primary" : "success";
  return (
    <div
      className={`bg-${completeClass} h-1 rounded-full`}
      style={{ width: `${progress}%` }}
    ></div>
  );
}

Slider.propTypes = {
  progress: PropTypes.number.isRequired,
};
