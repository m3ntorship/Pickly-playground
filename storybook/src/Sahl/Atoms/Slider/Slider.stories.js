import React from "react";

import Slider from "./Slider";

const defaultExport = {
  title: "Sahl/Atoms/Slider",
  component: Slider,
  argTypes: {
    // Default without options
    // progress: { control: "range" },
    progress: {
      control: {
        type: "range",
        min: 0,
        max: 100,
        step: 2,
      },
    },
  },
};

export default defaultExport;

const Template = (args) => <Slider {...args} />;

export const Primary = Template.bind({});
Primary.args = {
  progress: 20,
};
