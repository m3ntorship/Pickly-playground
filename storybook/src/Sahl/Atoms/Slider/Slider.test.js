import React from "react";
import renderer from "react-test-renderer";
import Slider from "./Slider";

describe("Slider Component", () => {
  it("should change its width when passing different values to progress prop ", () => {
    const tree = renderer.create(<Slider progress={24} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should change its width when passing different values to progress prop ", () => {
    const tree = renderer.create(<Slider progress={85} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
