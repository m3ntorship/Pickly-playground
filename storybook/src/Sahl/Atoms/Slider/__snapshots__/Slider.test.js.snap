// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider Component should change its width when passing different values to progress prop  1`] = `
<div
  className="bg-primary h-1 rounded-full"
  style={
    Object {
      "width": "24%",
    }
  }
/>
`;

exports[`Slider Component should change its width when passing different values to progress prop  2`] = `
<div
  className="bg-primary h-1 rounded-full"
  style={
    Object {
      "width": "85%",
    }
  }
/>
`;
